bluefog/torch/mpi_ops.cc

```diff
@@ -338,205 +407,64 @@ int DoNeighborAllreduce(::torch::Tensor tensor, ::torch::Tensor output,
 
   auto callback_wrapper = GetCallbackWrapper(handle, timeline_ptr, op_name, tid);
 
-  std::vector<int> recv_neighbors;
-  for (auto kv : neighbor_weights)
-      recv_neighbors.push_back(kv.first);
-  std::sort(recv_neighbors.begin(), recv_neighbors.end());
-
+  std::map<int, double> src_weights_map;
```

Comment:
Call this src_weights_ordered_map for clarification?

Reply:
Done.